packages/cache/src/LocalCacheStorage.ts

```diff
@@ -31,13 +30,25 @@ export class LocalCacheStorage extends CacheStorage {
     });
 
     await Promise.all(
-      files.map(async file => {
-        await fs.mkdirp(path.dirname(path.join(this.cwd, file)));
-        await fs.copy(
-          path.join(localCacheFolder, file),
-          path.join(this.cwd, file)
-        );
-      })
+      files
+        .filter(async file => {
+          const src = path.join(localCacheFolder, file);
+          const dest = path.join(this.cwd, file);
+
+          if (fs.existsSync(dest)) {
+            const stats = await Promise.all([fs.stat(src), fs.stat(dest)]);
+            return stats[0].mtime !== stats[1].mtime;
```

Comment:
Are there cases where `mtime` might be equal, but the files are different?

Reply:
You can fake mtime indeed. Though I’m not sure how one can do that. I’m not too concerned with this  particular optimization. If you believe it would be too unstable I’d take it out for now because I’m not sure about the usefulness of a local cache at the moment.